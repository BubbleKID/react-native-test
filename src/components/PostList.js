import React from 'react';
import {
  Text, View, Image, TouchableOpacity, FlatList, ScrollView, StyleSheet,
} from 'react-native';
const defaultImage = require('../assets/images/news.png');

const PostListScreen = (props) => {
  const _renderItem = ({ item, index }) => (
    <TouchableOpacity
      activeOpacity={0.5}
      key={index}
      style={styles.item}
      onPress={() => props.navigation.navigate('Detail', { id: item.id })}
    >
      <Image
        source={item.type_img_mobile_big != null ? { uri: item.type_img_mobile_big } : defaultImage} 
        style={styles.image}
      />
      <View style={{ flex: 1 }}>
        <Text
          key={item.id}
          style={styles.text}
        >
          {item.title.rendered}
        </Text>
      </View>
    </TouchableOpacity>
  );

  const { postData } = props;
  return (
    <ScrollView style={styles.container}>
      <FlatList
        data={postData}
        renderItem={_renderItem}
        style={styles.flatList}
      />
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  image: {
    width: 75,
    height: 50,
    marginRight: 10,
  },
  text: {
    fontWeight: '400',
    fontSize: 16,
  },
  item: {
    padding: 15,
    justifyContent: 'space-between',
    flexDirection: 'row',
    borderRadius: 2,
    borderColor: '#ddd',
    borderBottomWidth: 1,
  },
  flatList: {
    marginRight: 15,
    marginLeft: 15,
    backgroundColor: '#fff',
    borderWidth: 1,
    borderRadius: 2,
    borderColor: '#ddd',
    borderTopWidth: 0,
    borderBottomWidth: 0,
  }
});

export default PostListScreen;
