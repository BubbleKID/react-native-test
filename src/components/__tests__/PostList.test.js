import 'react-native';
import React from 'react';
import PostList from '../PostList';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

it('renders correctly', () => {
  const tree = renderer.create(<PostList />).toJSON();

  expect(tree).toMatchSnapshot();
});

it(' _renderItem works correctly', ()=>{
  const data = [{id:1,title:{rendered:'123'}},{id:2,title:{rendered:'321'}}];
  const component = shallow(<PostList data={data} />).dive();
  console.log(component)
  //expect(component).toMatchSnapshot();
});
