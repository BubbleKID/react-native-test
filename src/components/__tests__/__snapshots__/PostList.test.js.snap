// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` _renderItem works correctly 1`] = `
<ScrollViewMock
  style={
    Object {
      "flex": 1,
    }
  }
>
  <FlatList
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0,
        "borderColor": "#ddd",
        "borderRadius": 2,
        "borderTopWidth": 0,
        "borderWidth": 1,
        "marginLeft": 15,
        "marginRight": 15,
      }
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</ScrollViewMock>
`;

exports[`renders correctly 1`] = `null`;
